lock/client/client.go

```diff
@@ -0,0 +1,136 @@
+// Package client implements a client to the volmaster to acquire locks.
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/contiv/volplugin/config"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+// Driver is the main force behind the lock module, it controls all methods and
+// several variables.
+type Driver struct {
+	master         string
+	mountStopChans map[string]chan struct{}
+	mountMutex     sync.Mutex
```

Comment:
perhaps, but I really want a better reason to rewrite the patch than 
this. :) Maybe in a future refactor.

On 7 Mar 2016, at 11:40, Madhav Puri wrote:

>> +	"net/http"
>> +	"strings"
>> +	"sync"
>> +	"time"
>> +
>> +	"github.com/contiv/volplugin/config"
>> +
>> +	log "github.com/Sirupsen/logrus"
>> +)
>> +
>> +// Driver is the main force behind the lock module, it controls all 
>> methods and
>> +// several variables.
>> +type Driver struct {
>> +	master         string
>> +	mountStopChans map[string]chan struct{}
>> +	mountMutex     sync.Mutex
>
> minor: just wondering, if we shall just embed the `sync.Mutex` 
> directly here? That way we can just call `d.Lock()` and `d.Unlock()` 
> every, than more verbose `d.mountMutex...`
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/contiv/volplugin/pull/170/files#r55259202